File: frontend/src/lib/types/edit.ts
export interface CellChange {
  columnName: string
  primaryKey: Record<string, unknown>
  newValue: unknown
}

export interface UpdateCellsBatchRequest {
  connectionId: string
  tableName: string
  changes: CellChange[]
}

export interface UpdateCellsBatchResult {
  updatedCount: number
}

export interface PendingEdit {
  columnName: string
  primaryKey: Record<string, unknown>
  originalValue: unknown
  newValue: unknown
}

File: internal/model/edit.go
package model

const MaxBatchCellUpdates = 200

type CellChange struct {
	ColumnName string         `json:"columnName"`
	PrimaryKey map[string]any `json:"primaryKey"`
	NewValue   any            `json:"newValue"`
}

type UpdateCellsBatchRequest struct {
	ConnectionID string       `json:"connectionId"`
	TableName    string       `json:"tableName"`
	Changes      []CellChange `json:"changes"`
}

type UpdateCellsBatchResult struct {
	UpdatedCount int `json:"updatedCount"`
}
